lib/robot.js

```diff
@@ -144,19 +144,44 @@ class Robot {
       logger: log.child({installation: id})
     })
 
-    if (id) {
-      const res = await this.cache.wrap(`app:${id}:token`, () => {
-        log.trace(`creating token for installation`)
-        github.authenticate({type: 'integration', token: this.app()})
+    if (process.env.AUTH_METHOD === 'githubapp') {
+      if (id) {
+        const res = await this.cache.wrap(`app:${id}:token`, () => {
+          log.trace(`creating token for installation`)
+          github.authenticate({type: 'integration', token: this.app()})
 
-        return github.apps.createInstallationToken({installation_id: id})
-      }, {ttl: 60 * 59}) // Cache for 1 minute less than GitHub expiry
+          return github.apps.createInstallationToken({installation_id: id})
+        }, {ttl: 60 * 59}) // Cache for 1 minute less than GitHub expiry
 
-      github.authenticate({type: 'token', token: res.data.token})
+        github.authenticate({type: 'token', token: res.data.token})
+      } else {
+        github.authenticate({type: 'integration', token: this.app()})
+      }
     } else {
-      github.authenticate({type: 'integration', token: this.app()})
+      github = new GitHubApi({
+        debug: process.env.LOG_LEVEL === 'trace'
+      })
+      await this.cache.wrap(`app:oauth${process.env.CLIENT_ID}:token`, () => {
+        this.log.trace(`creating token for oauth client ${process.env.CLIENT_ID}`)
+        return github.authenticate({
+          type: 'oauth',
+          key: process.env.CLIENT_ID,
+          secret: process.env.CLIENT_SECRET
+        })
+      }, {
+        ttl: 60 * 60
+      })
+
+      /*
+      TODO: GitHubApi additional config for:
```

Comment:
@jbjonesjr this has been added by #331, so we just need to have it use the right config object.

Reply:
SWEEETTT!

Reply:
I don't realize undertand why this didn't get picked up when i merged master in... Or I guess i'm now behind master, and this what i'm behind. time to sync!